yb-voyager/src/callhome/diagnostics.go

```diff
@@ -0,0 +1,159 @@
+package callhome
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+//call-home json formats
+var (
+	jsonFilePath    string
+	Payload         payload
+	jsonBuf         []byte
+	SendDiagnostics bool
+)
+
+const(
+	CALL_HOME_SERVICE_HOST      = "34.83.149.226"
+	CALL_HOME_SERVICE_PORT      = 80
+)
+
+type payload struct {
+	MigrationUuid         uuid.UUID `json:"UUID"`
+	StartTime             string    `json:"start_time"`
+	LastUpdatedTime       string    `json:"last_updated_time"`
+	SourceDBType          string    `json:"source_db_type"`
+	SourceDBVersion       string    `json:"source_db_version"`
+	Issues                string    `json:"issues"`
+	DBObjects             string    `json:"database_objects"`
+	TargetDBVersion       string    `json:"target_db_version"`
+	NodeCount             int       `json:"node_count"`
+	ParallelJobs          int       `json:"parallel_jobs"`
+	TotalRows             int64     `json:"total_rows"`
+	TotalSize             int64     `json:"total_size"`
+	LargestTableRows      int64     `json:"largest_table_rows"`
+	LargestTableSize      int64     `json:"largest_table_size"`
+	TargetClusterLocation string    `json:"target_cluster_location"` //TODO
+	TargetDBCores         int       `json:"target_db_cores"`         //TODO
+	SourceCloudDBType     string    `json:"source_cloud_type"`       //TODO
+}
+
+// Fill in primary-key based fields, if needed
+func initJSON(exportdir string) {
+
+	jsonFilePath = filepath.Join(exportdir, "metainfo", "diagnostics.json")
+	file, err := os.OpenFile(jsonFilePath, os.O_RDWR|os.O_CREATE, 0644)
+	file.Close()
+	if err != nil {
+		log.Errorf("Error while creating/opening diagnostics.json file: %v", err)
+		return
+	}
+	jsonBuf, err = os.ReadFile(jsonFilePath)
+	if err != nil {
+		log.Errorf("Error while reading diagnostics.json file: %v", err)
+		return
+	}
+
+	if len(jsonBuf) != 0 {
+		err = json.Unmarshal(jsonBuf, &Payload)
+		if err != nil {
+			log.Errorf("Invalid diagnostics.json file: %v", err)
+			return
+		}
+	}
+
+	if Payload.MigrationUuid == uuid.Nil {
+		Payload.MigrationUuid, err = uuid.NewUUID()
+		Payload.StartTime = time.Now().Format("2006-01-02 15:04:05")
+		if err != nil {
+			log.Errorf("Error while generating new UUID for diagnostics.json: %v", err)
+			return
+		}
+	}
+
+}
+
+// Getter method for updating payload
+func GetPayload(exportDir string) *payload {
+	initJSON(exportDir)
```

Comment:
`initJSON()` should get called only once irrespective of how many times `GetPayload()` is called.
Call `initJSON()` only if `Payload.MigrationUuid == uuid.Nil`.

Reply:
GetPayload() is always called once and no more for each phase of the migration, can still make this change 👍🏽

Reply:
It is better to add the suggested check so that the code will continue working even if `GetPayload()` is called multiple times (in future).